Nl thermal

GZingaro opened the pull request:
## What? The convergence rate of the NR algorithm is not quadratic for linear elements for the Abaqus benchmark problem. 
## Why? Ensure NLTH1 is converging as expected before merging into develop branch. 
## How? Check into the implementation of nonlinear transient heat diffusion (NLTH1). 
## Testing? Evaluated using Abaqus Benchmark problem in Meeting Notes. Uploaded NLTH1_ABAQUS_Config.m and NLTH1_ABAQUS_Mesh.xlsx files to the configuration folder. 
## Anything Else? N/A

GZingaro committed:
Updated function points for TH3 - Thermal Nonlinear Diffusion.

GZingaro committed:
Computation for Jacobian matrix for nonlinear diffusion problem. Please refer to meeting notes for more detail.

GZingaro committed:
New function created to compute the constitutive law for nonlinear diffusion problem.

GZingaro committed:
Add manufacture solution for nonlinear diffusion to runtests.

GZingaro committed:
Modified Control.alpha in DiffusionTest for consistency.

GZingaro committed:
Modified function name for constitutive law for nonlinear heat diffusion problem to getD_NLTH1. The function title and inputs have been modified.

GZingaro committed:
Modified function name of getK_TH3 to getK_NLTH1 for the transient nonlinear heat diffusion problem. The function title and inputs have been modified. Please note the modified variable name of H for the Hessian matrix and beta for Control.alpha (time integration coefficient).

GZingaro committed:
Modified model name for transient nonlinear heat diffusion to NLTH1.

GZingaro committed:
Files used for testing of NLTH1 branch for Abaqus benchmark solution.

GZingaro committed:
Temporarily add plotting functionality in main file for debugging.

GZingaro committed:
Modified convergence rate computation.

GZingaro committed:
Modified value of heat conductivity. Please check convergence.

GZingaro committed:
Merge develop into NL-Thermal.

GZingaro committed:
Commented out the debugging tools to plot the residual of the NR over iterations.

GZingaro committed:
Fixed conflict associated with the material models of the NLTH1 and THLE1.

GZingaro committed:
Fixed RunTest conflicts when merging develop into NL-Thermal.

GZingaro committed:
Modified FileList in RunSim for Abaqus Nonlinear transient heat diffusion test case.

SaeedHatefi committed:
Modify getK function for nonlinear thermal diffusion

SaeedHatefi committed:
Add nonlinear thermal transient material model to all config files

SaeedHatefi committed:
Chanhe the name of config file for NLTH

SaeedHatefi committed:
Modify function header for nonlinear thermal transient

SaeedHatefi reviewed (changes requested):
- [x] Reviewed function(s): RunSim, RunTests, NLThermal1, main_nonlinear, write2vtk_dfsn, setMaterialModel, getK_NLTH1, getFlux_TH1, getD_NLTH1
- [x] Ran test case(s): RunSim, RunTests
* Purpose of the code: To add a nonlinearity feature to the code of thermal transient by solving a benchmark from ABAQUS
* Notes on implementation: The code works well for the current setup of the config file. However, there are some suggestions and issues needed to be addressed before merging:
   *  Consider to test the code for the cases mentioned in the Suggested test cases section.
   *  For the case of orthotropic, there is an error when it is tried to run the code. Also, for this case, getFlux function should be modified based on the conductivity in x and y direction
* Notes on legibility and style:
   * Consider to move NLTH1_ABAQUS_Mesh.xlsx file to Mesh Files folder
   * It seems that the files getK_TH3.m and getD_TH3.m should be removed since they are not being used in the code.
   * Suggest that to define a feval function for the definition of Constitutive Law in getK_NLTH1 instead of direct call of the fucntion getD_NLTH1. Function getK_TR2 can be a good reference for this.
   * Suggest to have a comment about the options we can have for the definition of getFlux: 1: We can define another getFlux for nonlinear thermal case or 2: we can modify the current getFlux_TH1 function.
   * In getFlux_TH1, I suggest to define a feval function instead of directly calling getD_TH1. For example, getStrain_ST1 is using feval function.
   * In getK_NLTH1, consider to rename the parameter beta to alpha because alpha is a time integration parameter. By this change, the material parameter should be rename to another thing.
   * In main_nonlinear.m, there are some debugging lines for plotting the residual values. It is a great job. I suggest to remove these lines after testing the code for the metioned tests
* Notes on maintainability: N/A
* Suggested test cases: N/A
   *  Consider to test the code for the case that we have body force. The goal is to check the residual force definition and the convergence rate for this case.
   *  Consider to test the code for the case that we have extenrnal flux. So, we will have a Neumann BC in this case. The goal is to check the residual force definition and the convergence rate for this case.
   *  Consider to test the code for Orthotopic case when the conductivity in x and y directions are different. In this case, we should have some modifications for getFlux function.

GZingaro committed:
Moved mesh for NLTH1 ABAQUS test cases into Mesh Files Folder.

GZingaro committed:
Add function handles for getD_NLTH1 in the getK_NLTH1.

GZingaro committed:
Deleted getK_TH3 as model was renamed to NLTH1 for nonlinear transient diffusion.

GZingaro committed:
Renamed linear diffusivity coefficient to k, to avoid confusion with the numerical integration parameter alpha.

GZingaro committed:
Modified field to solve bug for orthotropic diffusion.

GZingaro committed:
Removed debugging tools in main_nonlinear.

GZingaro committed:
Updated SetMaterialModel to account for the new getFlux_NLTH1, which modifies the inputs to the getD_NLTH1.

GZingaro committed:
Moved Abaqus mesh to mesh files folder.

GZingaro committed:
Created getFlux_NLTH1 for a seperate flux function for the NLTH1 model. Inputs of GetD_NLTH1 changes from GetD_TH1 as the temperature at the quadrature point is required.

GZingaro committed:
Updated description for inputs of nonlinear transient diffusion mode.

GZingaro commented:
> * [x]  Reviewed function(s): RunSim, RunTests, NLThermal1, main_nonlinear, write2vtk_dfsn, setMaterialModel, getK_NLTH1, getFlux_TH1, getD_NLTH1
> * [x]  Ran test case(s): RunSim, RunTests
> 
> * Purpose of the code: To add a nonlinearity feature to the code of thermal transient by solving a benchmark from ABAQUS
> * Notes on implementation: The code works well for the current setup of the config file. However, there are some suggestions and issues needed to be addressed before merging:
>   
>   * Consider to test the code for the cases mentioned in the Suggested test cases section. UPDATED: Ran manufactured solution for orthotropic transient diffusion. Code has been verified. Remaining tasks: run simulation with Neumann boundary conditions, and verify the heat flux. 

>   * For the case of orthotropic, there is an error when it is tried to run the code. Also, for this case, getFlux function should be modified based on the conductivity in x and y direction. UPDATE: Updated field "alpha2". Runs correctly and has been pushed. 
> * Notes on legibility and style:
>   
>   * Consider to move NLTH1_ABAQUS_Mesh.xlsx file to Mesh Files folder. UPDATE: Complete.
>   * It seems that the files getK_TH3.m and getD_TH3.m should be removed since they are not being used in the code. UPDATE: Complete.
>   * Suggest that to define a feval function for the definition of Constitutive Law in getK_NLTH1 instead of direct call of the fucntion getD_NLTH1. Function getK_TR2 can be a good reference for this. UPDATED: Complete. 
>   * Suggest to have a comment about the options we can have for the definition of getFlux: 1: We can define another getFlux for nonlinear thermal case or 2: we can modify the current getFlux_TH1 function. UPDATED: Created and modified GetFlux_TH1 for GetFlux_NLTH1. Verify is running correctly as the inputs to GetD in the function have been modified.
>   * In getFlux_TH1, I suggest to define a feval function instead of directly calling getD_TH1. For example, getStrain_ST1 is using feval function. UPDATE: Complete. 
>   * In getK_NLTH1, consider to rename the parameter beta to alpha because alpha is a time integration parameter. By this change, the material parameter should be rename to another thing. UPDATE: Renamed to k. 
>   * In main_nonlinear.m, there are some debugging lines for plotting the residual values. It is a great job. I suggest to remove these lines after testing the code for the mentioned tests. UPDATE: Complete.
> * Notes on maintainability: N/A
> * Suggested test cases: N/A
>   
>   * Consider to test the code for the case that we have body force. The goal is to check the residual force definition and the convergence rate for this case. UPDATE: Complete. 
>   * Consider to test the code for the case that we have extenrnal flux. So, we will have a Neumann BC in this case. The goal is to check the residual force definition and the convergence rate for this case. UPDATE: Ongoing task. 
>   * Consider to test the code for Orthotopic case when the conductivity in x and y directions are different. In this case, we should have some modifications for getFlux function. UPDATED: Complete.

GZingaro committed:
Switch the linear diffusivity constant name to "a" to avoid any confusion with other variables in the code

GZingaro committed:
Switch the linear diffusivity constant name to "a" to avoid any confusion with other variables in the code

GZingaro committed:
Removed input for manufactured solution.

GZingaro commented:
What? Requesting a review of the nonlinear transient thermal branch. 

Why? Check the implementation of the nonlinear transient thermal branch before merging into the develop branch.

How? Check into the implementation of nonlinear transient heat diffusion (NLTH1).

Testing? All previous comments have been addressed. Please run a nonlinear diffusion test with Neumann boundary conditions to verify the code is running correctly. Additionally, please check the getFlux_NLTH1 as the inputs to the constitutive law have been modified since the last revision. 

Anything Else? N/A

SaeedHatefi committed:
Move mesh file to the mesh folder
